modules/utils/tests/string_utils_tests.cpp

```diff
@@ -2,66 +2,135 @@
+                                                    .str = "/path/to/some/file.txt",
+                                                    .start_token = "aaa",
+                                                    .end_token = ".txt",
+                                                    .include_end_token = false,
+                                                    .expected = "/path/to/some/file" },
+                                          TestCase{ .description = "Truncate end invalid",
+                                                    .str = "/path/to/some/file.txt",
+                                                    .start_token = "some",
+                                                    .end_token = "bbb",
+                                                    .include_end_token = true,
+                                                    .expected = "some/file.txt" },
+                                          TestCase{ .description = "Truncate start and end empty",
+                                                    .str = "/path/to/some/file.txt",
+                                                    .start_token = "",
+                                                    .end_token = "",
+                                                    .include_end_token = true,
+                                                    .expected = "/path/to/some/file.txt" } };
+
+  for (const auto& test_case : TEST_CASES) {
     const auto truncated =
         truncate(test_case.str, test_case.start_token, test_case.end_token, test_case.include_end_token);
     EXPECT_EQ(truncated, test_case.expected) << test_case.description;
   }
+
+  constexpr auto CONSTEXPR_TEST_CASE = TEST_CASES[0];
+  constexpr auto CONSTEXPR_TRUNCATED =
+      truncate(CONSTEXPR_TEST_CASE.str, CONSTEXPR_TEST_CASE.start_token, CONSTEXPR_TEST_CASE.end_token,
+               CONSTEXPR_TEST_CASE.include_end_token);
+  EXPECT_EQ(CONSTEXPR_TRUNCATED, CONSTEXPR_TEST_CASE.expected) << CONSTEXPR_TEST_CASE.description;
+}
+
+TEST(StringUtilsTests, toUpperCase) {
+  const std::string test_string = "aNy_TEST_CaSe_42!";
+  const auto upper_case = toUpperCase(test_string);
+
+  EXPECT_EQ(upper_case, "ANY_TEST_CASE_42!");
 }
-}  // namespace heph::utils::tests
+
+TEST(StringUtilsTests, toUpperCaseRandom) {
+  auto mt = std::mt19937_64(std::random_device{}());
+  const auto random_string = randomString(mt);
+  const auto upper_case = toUpperCase(random_string);
+
+  for (const auto& c : upper_case) {
+    if (std::isalpha(c) != 0) {
+      EXPECT_TRUE(std::isupper(c) != 0);
+    }
+  }
+}
+
+TEST(StringUtilsTests, toSnakeCase) {
+  const std::string camel_case = "snakeCaseTest42!";
+  const auto snake_case = toSnakeCase(camel_case);
+
+  EXPECT_EQ(snake_case, "snake_case_test42!");
+
+  auto mt = std::mt19937_64(std::random_device{}());
+  const auto random_string = randomString(mt);
+  EXPECT_NO_THROW(static_cast<void>(toSnakeCase(random_string)));
```

Comment:
what is the point of this test? `toSnakeCase` never throws by definition

Reply:
I wasn't sure what happens if I input a non-alphabetic character into toUpper or toLower, so I added the test. It requires that all case conversions ignore non-alphabetic characters, even if the code is changed.

Reply:
toupper simply returns input if it cannot upper it so I don't think it is needed

Reply:
if you want to keep it I would check that the size is > 0 or actually > than `random_string` size.

Reply:
Probably best to remove it. Else I start testing for random stuff.